plugins/faucet/faucet.go

```diff
@@ -0,0 +1,125 @@
+package faucet
+
+import (
+	"context"
+	"crypto/ed25519"
+	"os"
+	"time"
+	"unsafe"
+
+	"github.com/iotaledger/goshimmer/client/wallet"
+	"github.com/iotaledger/goshimmer/client/wallet/packages/address"
+	"github.com/iotaledger/goshimmer/client/wallet/packages/seed"
+	walletseed "github.com/iotaledger/goshimmer/client/wallet/packages/seed"
+	"github.com/iotaledger/goshimmer/client/wallet/packages/sendoptions"
+	"github.com/iotaledger/goshimmer/packages/app/faucet"
+	"github.com/iotaledger/goshimmer/packages/core/ledger/vm/devnetvm"
+	"github.com/iotaledger/hive.go/core/bitmask"
+	"github.com/iotaledger/hive.go/core/marshalutil"
+)
+
+var (
+	maxTxBookedAwaitTime = 5 * time.Second
+)
+
+// remainder stays on index 0
+type Faucet struct {
+	*wallet.Wallet
+}
+
+// NewFaucet creates a new Faucet instance.
+func NewFaucet(faucetSeed *seed.Seed, walletStates string) *Faucet {
+	seed, lastAddressIndex, spentAddresses, assetRegistry, err := importWalletStateFile(walletStates)
```

Comment:
You don't need to dump the wallet file. Wallet should always be initialized from seed.